Compiling with HXCPP_MINGW on Windows gives bad assembly registers

While trying to compile `GcRegCapture.cpp` using mingw, the assembler gives the following error:
```
Error: C:\Users\resto\AppData\Local\Temp\ccKoOCub.s: Assembler messages:
C:\Users\resto\AppData\Local\Temp\ccKoOCub.s:21: Error: bad register name `%rbx'
C:\Users\resto\AppData\Local\Temp\ccKoOCub.s:29: Error: bad register name `%rbp'
C:\Users\resto\AppData\Local\Temp\ccKoOCub.s:37: Error: bad register name `%rdi'
C:\Users\resto\AppData\Local\Temp\ccKoOCub.s:45: Error: bad register name `%r12'
C:\Users\resto\AppData\Local\Temp\ccKoOCub.s:53: Error: bad register name `%r13'
C:\Users\resto\AppData\Local\Temp\ccKoOCub.s:61: Error: bad register name `%r14'
C:\Users\resto\AppData\Local\Temp\ccKoOCub.s:69: Error: bad register name `%r15'
```

I suspect that the error comes from these lines on `GcRegCapture.cpp`
```
      asm ("movq %%rbx, %0\n\t" : "=r" (regBx) );
      asm ("movq %%rbp, %0\n\t" : "=r" (regBp) );
      asm ("movq %%rdi, %0\n\t" : "=r" (regDi) );
      asm ("movq %%r12, %0\n\t" : "=r" (reg12) );
      asm ("movq %%r13, %0\n\t" : "=r" (reg13) );
      asm ("movq %%r14, %0\n\t" : "=r" (reg14) );
      asm ("movq %%r15, %0\n\t" : "=r" (reg15) );
```
 Perhaps changing `movq %%key` to `movq %key` when using mingw should work, as seen on this post from stack overflow -> https://stackoverflow.com/questions/20582058/error-bad-register-name-rax-while-compiling-with-mingw-on-64-bit-windows-7

Tried changing `movq %%key` to `movq %key` and got a different error. still no idea what to do hahah
```bat
Error: C:/HaxeToolkit/haxe/lib/hxcpp/git/src/hx/gc/GcRegCapture.cpp: In function 'void hx::CaptureX64(hx::RegisterCaptureBuffer&)':
C:/HaxeToolkit/haxe/lib/hxcpp/git/src/hx/gc/GcRegCapture.cpp:69:48: error: invalid 'asm': operand number missing after %-letter
       asm ("movq %rbx, %0\n\t" : "=r" (regBx) );
                                                ^
C:/HaxeToolkit/haxe/lib/hxcpp/git/src/hx/gc/GcRegCapture.cpp:70:48: error: invalid 'asm': operand number missing after %-letter
       asm ("movq %rbp, %0\n\t" : "=r" (regBp) );
                                                ^
C:/HaxeToolkit/haxe/lib/hxcpp/git/src/hx/gc/GcRegCapture.cpp:71:48: error: invalid 'asm': operand number missing after %-letter
       asm ("movq %rdi, %0\n\t" : "=r" (regDi) );
                                                ^
C:/HaxeToolkit/haxe/lib/hxcpp/git/src/hx/gc/GcRegCapture.cpp:72:48: error: invalid 'asm': operand number out of range
       asm ("movq %r12, %0\n\t" : "=r" (reg12) );
                                                ^
C:/HaxeToolkit/haxe/lib/hxcpp/git/src/hx/gc/GcRegCapture.cpp:73:48: error: invalid 'asm': operand number out of range
       asm ("movq %r13, %0\n\t" : "=r" (reg13) );
                                                ^
C:/HaxeToolkit/haxe/lib/hxcpp/git/src/hx/gc/GcRegCapture.cpp:74:48: error: invalid 'asm': operand number out of range
       asm ("movq %r14, %0\n\t" : "=r" (reg14) );
                                                ^
C:/HaxeToolkit/haxe/lib/hxcpp/git/src/hx/gc/GcRegCapture.cpp:75:48: error: invalid 'asm': operand number out of range
       asm ("movq %r15, %0\n\t" : "=r" (reg15) );
                                                ^
```
i have no idea what that means!
i love assembly

--- a/toolchain/mingw-toolchain.xml
+++ b/toolchain/mingw-toolchain.xml
@@ -43,6 +43,7 @@
   <cppflag value="-std=c++11" if="HXCPP_CPP11" />
   <cppflag value="-U__STRICT_ANSI__" />
   <flag value="-m32" unless="HXCPP_M64"/>
+  <flag value="-m64" if="HXCPP_M64"/>
   <flag value="-DHXCPP_M64" if="HXCPP_M64"/>
   <flag value="-I${HXCPP}/include"/>
   <objdir value="obj/mingw${OBJEXT}/"/>
@@ -56,6 +57,7 @@
   <flag value="-debug" if="debug"/>
   <flag value="--enable-auto-import"/>
   <flag value="-m32" unless="HXCPP_M64"/>
+  <flag value="-m64" if="HXCPP_M64"/>
   <flag value="-static" if="no_shared_libs"/>
   <flag value="-static-libgcc" if="no_shared_libs"/>
   <flag value="-static-libstdc++" if="no_shared_libs"/>
@@ -70,6 +72,7 @@
   <flag value="-Wl,--enable-auto-import"/>
   <flag value="-mwindows" if="SUBSYSTEMWINDOWS" />
   <flag value="-m32" unless="HXCPP_M64"/>
+  <flag value="-m64" if="HXCPP_M64"/>
   <flag value="-static" if="no_shared_libs"/>
   <flag value="-static-libgcc" if="no_shared_libs" unless="linux_host"/>
   <flag value="-static-libstdc++" if="no_shared_libs" unless="linux_host"/>

